src/gui/EffectView.cpp

```diff
@@ -45,65 +49,60 @@
 namespace lmms::gui
 {
 
-EffectView::EffectView( Effect * _model, QWidget * _parent ) :
-	PluginView( _model, _parent ),
-	m_bg( embed::getIconPixmap( "effect_plugin" ) ),
-	m_subWindow( nullptr ),
+EffectView::EffectView(Effect * _model, QWidget * _parent) :
+	PluginView(_model, _parent),
+	m_subWindow(nullptr),
 	m_controlView(nullptr),
 	m_dragging(false)
 {
-	setFixedSize(EffectView::DEFAULT_WIDTH, EffectView::DEFAULT_HEIGHT);
 	setFocusPolicy(Qt::StrongFocus);
+	setViewWidth(EffectView::DEFAULT_WIDTH);
 
-	// Disable effects that are of type "DummyEffect"
+	m_mainLayout = new QHBoxLayout();
+	m_mainLayout->setContentsMargins(8, 2, 8, 2);
+
+	bool hasControls = effect()->controls()->controlCount() > 0;
 	bool isEnabled = !dynamic_cast<DummyEffect *>( effect() );
-	m_bypass = new LedCheckBox( this, "", isEnabled ? LedCheckBox::LedColor::Green : LedCheckBox::LedColor::Red );
-	m_bypass->move( 3, 3 );
-	m_bypass->setEnabled( isEnabled );
 
+	m_bypass = new LedCheckBox(this, "", isEnabled ? LedCheckBox::LedColor::Green : LedCheckBox::LedColor::Red);
+	m_bypass->setEnabled(isEnabled);
 	m_bypass->setToolTip(tr("On/Off"));
+	m_mainLayout->addWidget(m_bypass);
+
+	QFont labelFont = adjustedToPixelSize(font(), 10);
+	m_label = new EffectLabelButton(this, this);
+	m_label->setText(model()->displayName());
+	m_label->setFont(labelFont);
+	m_label->setCheckable(true);
+	m_label->setSizePolicy(QSizePolicy::MinimumExpanding, QSizePolicy::Maximum);
+	if(hasControls)
+	{
+		connect(m_label, SIGNAL(clicked()), this, SLOT(editControls()));
+	}
+	m_mainLayout->addWidget(m_label);
 
+	m_wetDry = new Knob(KnobType::Small17, this);
+	m_wetDry->setEnabled(isEnabled);
+	m_wetDry->setHintText(tr("Wet Level:"), "");
+	m_mainLayout->addWidget(m_wetDry);
 
-	m_wetDry = new Knob( KnobType::Bright26, this );
-	m_wetDry->setLabel( tr( "W/D" ) );
-	m_wetDry->move( 40 - m_wetDry->width() / 2, 5 );
-	m_wetDry->setEnabled( isEnabled );
-	m_wetDry->setHintText( tr( "Wet Level:" ), "" );
-
-
-	m_autoQuit = new TempoSyncKnob( KnobType::Bright26, this );
-	m_autoQuit->setLabel( tr( "DECAY" ) );
-	m_autoQuit->move( 78 - m_autoQuit->width() / 2, 5 );
-	m_autoQuit->setEnabled( isEnabled && !effect()->m_autoQuitDisabled );
-	m_autoQuit->setHintText( tr( "Time:" ), "ms" );
-
-
-	m_gate = new Knob( KnobType::Bright26, this );
-	m_gate->setLabel( tr( "GATE" ) );
-	m_gate->move( 116 - m_gate->width() / 2, 5 );
-	m_gate->setEnabled( isEnabled && !effect()->m_autoQuitDisabled );
-	m_gate->setHintText( tr( "Gate:" ), "" );
-
+	m_autoQuit = new TempoSyncKnob(KnobType::Small17, this);
+	m_autoQuit->setEnabled(isEnabled && !effect()->m_autoQuitDisabled);
+	m_autoQuit->setVisible(isEnabled && !effect()->m_autoQuitDisabled);
+	m_autoQuit->setHintText(tr("Stop after:"), "ms");
+	m_mainLayout->addWidget(m_autoQuit);
 
-	setModel( _model );
+	setModel(_model);
```

Comment:
```suggestion
	setModel( _model );
```
Put in those spaces.